spharpy/spherical.py

```diff
@@ -1115,3 +1118,385 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        if value % 1 != 0:
+            raise ValueError("n_max must be an integer value")
+        if self.channel_convention == "fuma" and value > 3:
+            raise ValueError("n_max > 3 is not allowed with 'fuma' "
+                             "channel convention")
+        if self.normalization == "maxN" and value > 3:
+            raise ValueError("n_max > 3 is not allowed with 'maxN' "
+                             "normalization")
+        if int(value) != self._n_max:
+            self._reset_compute_attributes()
+            self._n_max = int(value)  # Cast to int for safety
+
+    @property
+    def coordinates(self):
+        """Get the coordinates object."""
+        return self._coordinates
+
+    @coordinates.setter
+    def coordinates(self, value):
+        """Set the coordinates object."""
+        if not isinstance(value, pf.Coordinates):
+            raise TypeError("coordinates must be a pyfar.Coordinates "
+                            "object or spharpy.SamplingSphere object")
+        if value.cdim != 1:
+            raise ValueError("Coordinates must be a 1D array")
+        if value.csize == 0:
+            raise ValueError("Coordinates cannot be empty")
+        if value != self._coordinates:
+            self._reset_compute_attributes()
+            self._coordinates = value
+
+    @property
+    def basis_type(self):
+        """Get or set the type of spherical harmonic basis."""
+        return self._basis_type
+
+    @basis_type.setter
+    def basis_type(self, value):
+        """
+        Set the type (real or complex) of spherical harmonic basis.
+        """
+        if value not in ["complex", "real"]:
+            raise ValueError("Invalid basis type, only "
+                             "'complex' and 'real' are supported")
+        if value != self._basis_type:
+            self._reset_compute_attributes()
+            self._basis_type = value
+
+    @property
+    def inverse_method(self):
+        """Get or set the type of inverse transform."""
+        return self._inverse_method
+
+    @inverse_method.setter
+    def inverse_method(self, value):
+        """Set the inverse transform type."""
+        allowed = ["pseudo_inverse", "quadrature", "auto"]
+        if value not in allowed:
+            raise ValueError("Invalid inverse method. Allowed values are 'inverse', "
+                             "'quadrature', or 'auto'.")
+        # If using simple pf.Coordinates and not SamplingSphere, 'auto' is not allowed
+        if not isinstance(self.coordinates, SamplingSphere) and value == "auto":
+            raise ValueError("inverse_method='auto' is not allowed without SamplingSphere coordinates. " \
+            "Please specify 'pseudo_inverse' or 'quadrature'.")
+        # If using quadrature, weights must be set
+        if value == "quadrature" and self.coordinates.weights is None:
+            raise ValueError("Weights must be set for quadrature inverse method")
```

Comment:
You can use the new property now:
```suggestion
        if not hasattr(self, 'quadrature') or not self.quadrature:
            raise ValueError("'quadrature' requires coordinates in a SamplingSphere object and SampülingSphere.quadrature = True")
```